src/main/java/net/minecraftforge/event/world/BlockEvent.java

```diff
@@ -282,6 +283,44 @@ public CreateFluidSourceEvent(World world, BlockPos pos, IBlockState state)
             super(world, pos, state);
         }
     }
+
+    /**
+     * Fired when liquids mix to create a new block.
+     */
+    public static class FluidMixEvent extends BlockEvent
+    {
+        private final Material liquid1;
+        private final Material liquid2;
+        private IBlockState block;
+
+        public FluidMixEvent(World world, BlockPos pos, Material liquid1, Material liquid2, IBlockState block)
+        {
+            super(world, pos, block);
+            this.liquid1 = liquid1;
+            this.liquid2 = liquid2;
+            this.block = block;
+        }
+
+        public Material getLiquid1()
+        {
+            return liquid1;
+        }
+
+        public Material getLiquid2()
+        {
+            return liquid2;
+        }
+
+        @Override
+        public IBlockState getState()
```

Comment:
For consistency, this should probably be called `getBlockState`, the field can be `blockState`, and the setter can be `setBlockState`.

Reply:
I used FluidMixEvent::getState in hopes of preventing the casual modder from accessing BlockEvent::getState which I have rendered irrelevant.

Reply:
What about `mixState` then?

Reply:
Renamed it to `newState`, `getNewState` and `setNewState`.